pandas/core/indexes/datetimes.py

```diff
@@ -175,6 +175,7 @@ class DatetimeIndex(DatetimeArrayMixin, DatelikeOps, TimelikeOps,
     pandas.to_datetime : Convert argument to datetime
     """
     _resolution = cache_readonly(DatetimeArrayMixin._resolution.fget)
+    _shallow_copy = Index._shallow_copy
```

Comment:
why do you need to define this? (or is this a mix-in)?

Reply:
Same as in the TimedeltaIndex PR; we need to get the Index version specifically until we get rid of the DatetimeLikeArrayMixin version in #23430.